### PR TITLE
CI: add workflow for uplaoding the docker image

### PR DESCRIPTION
**Pull request by ynfle**
@ErikSchierboom from my understanding, there is no need to upload ECR being as the docker images are built in the GitHub actions which have access to the docker hub images and this image wouldn't need to be stored on ECR.

I assume you'd want to add a CODEOWNERS file

Also, could think correct keys and stuff be added to this repo?

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+name: Push Docker images to Docker Hub 
+
+on:
+  push:
+    branches: [main, master]
```
Comment (ErikSchierboom) [inline .github/workflows/docker.yml:5]: You can simplify this to:

```suggestion
    branches: [main]
```

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@a66e35b9cbcf4ad0ea91ffcaf7bbad63ad9e0229 # 2.7.0
+        with:
+          context: .
+          file: ./Dockerfile
+          push: true
+          tags: |
+            ${{ github.event.repository.full_name }}:latest
+            ${{ github.event.repository.full_name }}:${{ github.sha }}
```
Comment (ErikSchierboom) [inline .github/workflows/docker.yml:41]: I assume you'll be using either latest or the sha when referring this Docker image? As in: no versions.

**Comment by ErikSchierboom** [comment]
> I assume you'd want to add a CODEOWNERS file

Yes please. Could you add it to this PR or open a second PR?

> Also, could think correct keys and stuff be added to this repo?

@iHiD A job for you.

File: .github/workflows/docker.yml (hunk 2 above)
Comment (ynfle) [reply .github/workflows/docker.yml:41]: To be honest, I just copied straight out of the `nim-test-runner` one. I don't think we'll be needing version unless different tools want to run different versions of nim. For now I don't think there is a need to version.

Do you think it's useful to keep the sha version?

Comment (ErikSchierboom) [reply .github/workflows/docker.yml:41]: In this: not really, as we are in full control of the contents of the image.

**Comment by ynfle** [comment]
> > I assume you'd want to add a CODEOWNERS file
> 
> Yes please. Could you add it to this PR or open a second PR?

I just committed it directly

**Comment by ErikSchierboom** [comment]
Great, thanks

**Comment by iHiD** [comment]
Keys added for Dockerhub.

File: .github/workflows/docker.yml (hunk 2 above)
Comment (ynfle) [reply .github/workflows/docker.yml:41]: I think I'm gonna keep this as to be able to test for each tooling on each update of the base image and if something doesn't work it gives us the flexibility when needed.
It shouldn't pollute ECS or Docker Hub as there won't be many changes other optimization and/or nim upgrades.

**Comment by ErikSchierboom** [comment]
@ynfle I'll fix the Docker hub push error

**Comment by ynfle** [comment]
> @ynfle I'll fix the Docker hub push error

Which error?

**Comment by ErikSchierboom** [comment]
This one: https://github.com/exercism/nim-docker-base/actions/runs/1535087439

**Comment by ynfle** [comment]
I only saw that it error after I posted that. When I had checked it was still running 🤷‍♂️

**Comment by ynfle** [comment]
Did you upload it manually?
It's up on https://hub.docker.com/r/exercism/nim-docker-base.
Does it need to uploaded manually the first time?

**Comment by ErikSchierboom** [comment]
> Does it need to uploaded manually the first time?

That is correct. I then need to set permissions manually, after which the workflow works as intended.

**Comment by ErikSchierboom** [comment]
@ynfle It is now working: https://github.com/exercism/nim-docker-base/runs/4407933097?check_suite_focus=true

**Comment by ynfle** [comment]
> @ynfle It is now working: https://github.com/exercism/nim-docker-base/runs/4407933097?check_suite_focus=true

Ya I re-ran it to test. Thanks

**Pull request merged**